Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -432,6 +431,47 @@ class BrowserViewController: UIViewController {
         updateTabCountUsingTabManager(tabManager, animated: false)
     }
 
+    private func showCrashOptInAlert() {
+        let alert = UIAlertController.crashOptInAlert(
+            sendReportCallback: { _ in
+                // Turn on uploading but don't save opt-in flag to profile because this is a one time send.
+                configureActiveCrashReporter(true)
+                self.showRestoreTabsAlert()
+            },
+            alwaysSendCallback: { _ in
+                self.profile.prefs.setBool(true, forKey: "crashreports.send.always")
+                configureActiveCrashReporter(true)
+                self.showRestoreTabsAlert()
+            },
+            dontSendCallback: { _ in
+                // no-op: Do nothing if we don't want to send it
+                self.showRestoreTabsAlert()
+            }
+        )
+        self.presentViewController(alert, animated: true, completion: nil)
+    }
+
+    private func showRestoreTabsAlert() {
+        func addAndSelect() {
+            let tab = tabManager.addTab()
+            tabManager.selectTab(tab)
+        }
+
+        let alert = UIAlertController.restoreTabsAlert(
+            okayCallback: { _ in
+                self.restoreTabs()
+                if self.tabManager.count == 0 {
+                    addAndSelect()
```

Comment:
I've moved the restore tabs logic we have in the BVC to the restoreTabs method in tabManager